File: app/src/main/java/com/neome/java/api/ent/base/dto/DtoDebuggerLogEntry.java
// neome.ai API - do not change
//

package com.neome.java.api.ent.base.dto;

import com.neome.java.api.ent.base.Types.EnumLogType;
import com.neome.java.api.meta.base.dto.DefnForm;
import com.neome.java.api.meta.base.dto.DtoLogTree;
import com.neome.java.api.meta.base.dto.FormValueRaw;
import com.neome.java.api.nucleus.base.dto.EnvError;

import org.jetbrains.annotations.Nullable;

import java.util.Date;

@SuppressWarnings("unused")
public class DtoDebuggerLogEntry
{
  @Nullable
  public String caller;

  @Nullable
  public Date dateTime;

  @Nullable
  public EnvError envError;

  @Nullable
  public DefnForm inputForm;

  @Nullable
  public DtoLogTree inputFormLogTree;

  @Nullable
  public FormValueRaw inputFormValue;

  @Nullable
  public EnumLogType logType;

  @Nullable
  public String name;

  @Nullable
  public DefnForm outputForm;

  @Nullable
  public DtoLogTree outputFormLogTree;

  @Nullable
  public FormValueRaw outputFormValue;

  @Nullable
  public Object summary;
}

File: app/src/main/java/com/neome/api/ent/base/dto/DtoDebuggerLogEntry.kt
// neome.ai API - do not change
//
@file:Suppress("unused", "MemberVisibilityCanBePrivate", "CanBeParameter", "PropertyName")


package com.neome.api.ent.base.dto

import kotlinx.serialization.json.JsonElement
import com.neome.api.meta.base.dto.DefnForm
import com.neome.api.meta.base.dto.DtoLogTree
import com.neome.api.ent.base.Types.EnumLogType
import com.neome.api.nucleus.base.dto.EnvError
import com.neome.api.meta.base.dto.FormValueRaw

interface DtoDebuggerLogEntry
{
  val caller: String?
  val dateTime: String?
  val envError: EnvError?
  val inputForm: DefnForm?
  val inputFormLogTree: DtoLogTree?
  val inputFormValue: FormValueRaw?
  val logType: EnumLogType?
  val name: String?
  val outputForm: DefnForm?
  val outputFormLogTree: DtoLogTree?
  val outputFormValue: FormValueRaw?
  val summary: JsonElement?
}